### ISSUE TITLE
MW4 crash

### ISSUE THREAD

**Issue by ACMBakx** [mworion/MountWizzard4#191]
Last night I tried to build a new model.
I had the Scopedome disconnected and rotated the dome by hand.
This to speed up the process significantly.
I noticed during a first short trial of 25 points that the download of the images (as it appeared to me) took sometimes very long.
Most images came in quick and were solved ok, but some took a long time.
So the time gain by manually slewing the dome was actually zero.
I had 2.03 installed and upgraded to 2.04 on the last run. Maybe there was a bug in 2.03?
But that did not cure the problem.
Anyway, I wanted the 90 pts custom made model to run and started it off,
Again the download of the images was very random fast to slow and came finally to a complete halt.

maybe from the log you can tell what the problem is.

Regards
Arie
[Champignon90pts.zip](https://github.com/mworion/MountWizzard4/files/7252340/Champignon90pts.zip)
[mw4-2021-09-28.log](https://github.com/mworion/MountWizzard4/files/7252341/mw4-2021-09-28.log)

**Comment by mworion** [comment]
Hi Arie,
out of the log:
- I see many temporary socket timeouts for the connection between mount and pc. Anything happened in the connection? 2nd instance of MW4 running (load)? Wireless connection? This might explain some delays
- I've seen a bug in command stop exposure in MW4 -> this I is corrected in the 2.0.5 service release now
- You slew your dome manually and have better speed? Could you explain, why the slew from MW4 takes longer?
- The complete halt might be caused by the socket timeouts as well.
Michel

**Comment by ACMBakx** [comment]
I tried MC2 after this and that crashed too.
Martin tells me from the log that there is a communication problem with the mount that causes this.
You tell me something similar. Something to investigate.
I noticed that that when the mount finished a slew it took a unusual long time before I got the beep from the mount that it had finished the slew. That’s usually after a second or so. This time this took much longer.
What I meant with manually rotating the dome, I have the motor mechanically disengaged from the drive mechanism. Like that the dome rotates freely. Very light. Can do it with my finger tips. The mount slews a lot faster than the dome ever will, because of its mechanics. Like this a 90 pts model should only take 20-25 minutes only versus 1.5 x 2 hrs So this has nothing to do with MW4.
I did however beef-up the electric motor from 12v 100w to 12v 150w and a stronger powersupply unit from 12v 180w to 13.8v 320w.  With this the dome slewing is a lot more responsive.
Regards
Arie

**Comment by mworion** [comment]
Hi Arie,
dome topic understood. If I look to you points file, this is not optimal for dome use. Normally I should be possible to run a model with only one complete turn of the dome without changing direction or something else. MW4 with East West sorting the points and enable dome following feature should do the trick. I would be happy to check in which setting we get to model run time at least to 30-40 min without any manual work. 
Michel

**Comment by ACMBakx** [comment]
To minimize the dome slewing I have tried to follow the slot with the telescope.
Set the dome in a certain azimuth. Then take 5 shots with the telescope at increasing altitude. Say 5 shots.
Flip the telescope over and take another 5 shots coming back down again.
Slew the dome 15° and repeat the process for a full circle. Then the dome makes only 1 turn at 15° steps.
The result was an awful distribution of survey points. Some dense areas and some big voids.
So that did not work.
East/West sorting is perfect for a fork mounted telescope. But the GEM offset forces the dome to make extreme slew corrections at higher alt's  Sometimes way more than 90°. For the next point, which then sits at a low alt, the dome has to slew back again.
Especially these high alt points make the dome slew like crazy.
With my point model the dome makes only small slews for most points and the high alt ones aren't that many are taken as the last ones. Well, that's my theory. . . . . . . 
If you plot the dome positions along the survey points sequence, you should see these excessive slews at high alt telescope positions.
Add to that the Scopedome "unwind" feature, that causes the dome to slew at times "the long way around" instead of the shortest slew distance. This is caused by the limitation of the dome position encoder, which has a limited "max" value. When this will be exceeded with a slew, the dome rotates the other way around to reduce the encoder position value.
It is not as straight forward as it should be. I also guess that different domes use different control soft and hard ware, for simplicity.

**Comment by mworion** [comment]
Arie, 
I put that as a task who to optimize this especially for small domes. Larger ones should not be that influenced as the GEM offset might be smaller.
Closing the original issue, but definitely learn what happened regarding the connectivity issue as I might be able to detect such an issue and give some hints in the message window.
Michel

**Comment by ACMBakx** [comment]
Michael,
I found the cause of the problem: the camera!
The new Touptek2600Pro is having problems with downloading images to the PC. This became very eratic after some time.
I suppose this has confused MW4. The camera is send back to Lacerta for investigation.
I have tried it with different USB cables and sockets and on 3 different PC's. All the same. Camera is NBG.
Currently I have the Atik16200 installed again on my Epsilon. Because the Atik ascom driver is a real hopeless piece of .... and NINA has a  native driver for this camera that works and MC2 offers the option to image via NINA, I ran a 95 points model yesterday evening. This went perfect. Not a single point lost. Proving that the equipment in the dome is ok.
As soon as I have the opportunity I'll do a model-run using MW4 (latest release) and 'll let you know.
Arie
